### ISSUE TITLE
performance issues

### ISSUE THREAD

**Issue by hiban1** [borup3/Lighting-System-2D#18]
Im using this but the game is lagging, if there is too much lights I'm modyfying the global_lightning function but its not enought, I guess the shader is very recursfull

**Comment by hiban1** [comment]
Yes I've done it but it doesn't change anything, all I know is that it's comming from the lights because when my camera is in the shadow nothing happen, or maybe it's comming from my computer because I don't have a dedicated graphicall card but when I test the project on his own and not in my game it works well...

**Comment by DevTRUCKer** [comment]
hmm... yea I think the issue from your computer _**OR**_ I still also feel it's from the lighting system itself because it did not get updated for so long time so maybe it's not compatible a little with your GMS version, idk your GMS version, if you are using "Monthly" or "Beta" maybe try the "LTS" version in a separate project.

Also, I have a suggestion, and I know it still will not do anything but did you change the `in_TexCoord` to `in_TextureCoord` in the two shaders? check this (https://github.com/borup3/Lighting-System-2D/pull/7)

So yeah, I don't have any other suggestions the last thing I can tell you is... (nope not upgrade your PC) it is you can give me an example project with the same lights and shadows to see if my PC too can handle it.
Yep, that's it I hope you find a solution SeeYa.

**Comment by hiban1** [comment]
Ok I will try next time because I have a lot of things to do on my game. Hope I manage to do it, it will bring a lot...

**Comment by DevTRUCKer** [comment]
Suggest!
You can use beta with the new GMRT if you don't have but **Please backup your project first**.
GMRT focuses on Performance so your game may work better with it but if any errors show up you may work again on whatever version and runtime you were using.
